datalad/support/annexrepo.py

```diff
@@ -297,6 +297,21 @@ def add_remote(self, name, url, options=None):
                              " -S %s" % c.ctrl_path)
             writer.release()
 
+    def set_remote_url(self, name, url, push=False):
+        """Overrides method from GitRepo in order to set
+        remote.<name>.annex-ssh-options in case of a SSH remote."""
+
+        super(AnnexRepo, self).set_remote_url(name, url, push=push)
+        from datalad.support.network import is_ssh
+        if is_ssh(url):
+            c = ssh_manager.get_connection(url)
```

Comment:
I'm not sure, what exactly you are talking about.

1. What do you mean by "caught upstairs and handled"? If "upstairs" handles such an exception, how is this problematic? Or do you mean some `except Exception"? If this leads to an issue, I'd consider this except-clause the issue!

2. If `writer = self.repo.config_writer()` causes an Exception, then `writer.release()` will also fail (`AttributeError`)